### PR TITLE
feature/atom-with-sup-and-sub

### PR DESCRIPTION
**Pull request by builtbywill**
Correctly parse `x^2_2` the same way as `x_2^2`

### CODE REVIEW

**Comment by jkdufair** [comment]
@builtbywill Code looks reasonable. I have your branches fired up. I can't seem to find behavior different in `master` vs. this branch when creating an expression formula with `x^2_2` and `x_2^2`. Could you explain a good way to test this in Variate?

**Comment by builtbywill** [comment]
@jkdufair The bug was that with the subscript came after the superscript, it was completely lost, so `x^2_2` became `x^2` in sympy

So for example an answer of `x^2_1` would be counted as correct, since it too would become `x^2` in sympy

File: tests/all_good_test.py [hunk 1]
```diff
@@ -217,11 +217,11 @@ class TestAllGood(object):
         ("[a + b]", _Add(a, b)),
         ("\\frac{d}{dx} [ \\tan x ]", Derivative(tan(x), x)),
         ("2\\overline{x}", 2 * Symbol('xbar', real=True)),
-        ("2\\overline{x}_n", 2 * Symbol('xbar_{n}', real=True)),
-        ("\\frac{x}{\\overline{x}_n}", x / Symbol('xbar_{n}', real=True)),
-        ("\\frac{\\sin(x)}{\\overline{x}_n}", sin(Symbol('x', real=True)) / Symbol('xbar_{n}', real=True)),
+        ("2\\overline{x}_n", 2 * Symbol('xbar_n', real=True)),
```
Comment (lily514) [inline tests/all_good_test.py:220]: @builtbywill should one of these have a subscript that is more than one char long?

Comment (lily514) [reply tests/all_good_test.py:220]: {x}_{i,j} for example

**Comment by jkdufair** [comment]
@builtbywill this is working correctly for me. Interestingly, on `master`, `x^y_z` is not being evaluated the same as `x^y`. Not sure if that factors in here in any way or if there's a test case we need to consider. Since it's also not being evaluated the same after this branch :joy:. But there ya go. :thumps:

**Review by jkdufair — APPROVED** [review]

File: tests/all_good_test.py (hunk 1 above)
Comment (builtbywill) [reply tests/all_good_test.py:220]: @lily514 nice catch, yeah let me add some more tests

**Comment by builtbywill** [comment]
@lily514 @jkdufair @caseyveronica4 unit tests have been expanded, and had to make some tweaks to the grammar and logic

File: tests/atom_expr_test.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+]
+
+subscripts = [
+    ('2'),
+    ('{23}'),
+    ('i'),
+    ('{ij}'),
+    ('{i,j}'),
+    ('{good}'),
+    ('{x^2}')
```
Comment (builtbywill) [inline tests/atom_expr_test.py:20]: Basically now the atom's subscript latex is used as-is in the symbol name

**Review by lily514 — APPROVED** [review]

**Pull request merged**